resourseleak/TestConstructSimplestScenarioForLeak.java

```diff
@@ -0,0 +1,54 @@
+package resourceleak;
+
+import com.amazon.qtt.vault.VaultFile;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+
+public class TestConstructSimplestScenarioForLeak {
+
+    // MuGraph does not capture the context of whether the path to exit node is reached along return or throw statements.
+
+    @VisibleForTesting
+    void testConstructSimplestScenarioForLeak(VaultFile uploadRoot, String key) {
+        String testArtifactsMetadata = collectMetadataForTestArtifacts(uploadRoot);
+        File metadataFile = null;
+        OutputStreamWriter writer = null;
+        try {
+            metadataFile = File.createTempFile("_test_artifacts_metadata", ".json");
+            writer = new OutputStreamWriter(new FileOutputStream(metadataFile), Charsets.UTF_8.newEncoder());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileOutputStream*. Currently, there are execution paths that do not contain closure statements, for example, when Charsets.UTF_8.newEncoder throws an exception. Either a) close the object returned by *FileOutputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileOutputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).